flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/TableMaintenance.java

```diff
@@ -0,0 +1,318 @@
+                .process(
+                    new TriggerManager(
+                        loader,
+                        lockFactory,
+                        taskNames,
+                        evaluators,
+                        rateLimit.toMillis(),
+                        lockCheckDelay.toMillis()))
+                .name(TRIGGER_MANAGER_OPERATOR_NAME)
+                .uid(TRIGGER_MANAGER_OPERATOR_NAME + uidSuffix)
+                .slotSharingGroup(slotSharingGroup)
+                .forceNonParallel()
+                .assignTimestampsAndWatermarks(new PunctuatedWatermarkStrategy())
+                .name(WATERMARK_ASSIGNER_OPERATOR_NAME)
+                .uid(WATERMARK_ASSIGNER_OPERATOR_NAME + uidSuffix)
+                .slotSharingGroup(slotSharingGroup)
+                .forceNonParallel();
+
+        // Add the specific tasks
+        DataStream<TaskResult> unioned = null;
+        for (int i = 0; i < taskBuilders.size(); ++i) {
+          int finalIndex = i;
+          DataStream<Trigger> filtered =
+              triggers
+                  .filter(t -> t.taskId() != null && t.taskId() == finalIndex)
+                  .name(FILTER_OPERATOR_NAME_PREFIX + i)
+                  .forceNonParallel()
+                  .uid(FILTER_OPERATOR_NAME_PREFIX + i + "-" + uidSuffix)
+                  .slotSharingGroup(slotSharingGroup);
+          MaintenanceTaskBuilder<?> builder = taskBuilders.get(i);
+          DataStream<TaskResult> result =
+              builder.append(
+                  filtered, i, taskNames.get(i), loader, uidSuffix, slotSharingGroup, parallelism);
+          if (unioned == null) {
+            unioned = result;
+          } else {
+            unioned = unioned.union(result);
+          }
+        }
+
+        // Add the LockRemover to the end
+        unioned
+            .transform(
+                LOCK_REMOVER_OPERATOR_NAME,
+                TypeInformation.of(Void.class),
+                new LockRemover(lockFactory, taskNames))
+            .forceNonParallel()
+            .uid("lock-remover-" + uidSuffix)
+            .slotSharingGroup(slotSharingGroup);
+      }
+    }
+
+    private DataStream<TableChange> changeStream(TableLoader loader) {
+      if (inputStream == null) {
+        // Create a monitor source to provide the TableChange stream
+        MonitorSource source =
+            new MonitorSource(
+                loader, RateLimiterStrategy.perSecond(1.0 / rateLimit.getSeconds()), maxReadBack);
+        return env.fromSource(source, WatermarkStrategy.noWatermarks(), SOURCE_OPERATOR_NAME)
+            .uid(SOURCE_OPERATOR_NAME + uidSuffix)
+            .slotSharingGroup(slotSharingGroup)
+            .forceNonParallel();
+      } else {
+        return inputStream.global();
+      }
+    }
+  }
+
+  private static String nameFor(MaintenanceTaskBuilder<?> streamBuilder, int taskId) {
+    return String.format("%s [%d]", streamBuilder.getClass().getSimpleName(), taskId);
```

Comment:
why do we  use class name? can't we use the `name()` getter from `MaintenanceTaskBuilder`?

Reply:
This is the method we use to initialize the `name` attribute for the task.
This method is used to generate the names, like this:
```
      for (int i = 0; i < taskBuilders.size(); ++i) {
        taskNames.add(nameFor(taskBuilders.get(i), i));
        evaluators.add(taskBuilders.get(i).evaluator());
      }
```